imap_processing/lo/l1c/lo_l1c.py

```diff
@@ -0,0 +1,171 @@
+    return create_file_paths
+
+
+# TODO: This is going to work differently when I sample data.
+#  The data_fields input is temporary.
+def create_datasets(attr_mgr, logical_source, data_fields):
+    """Create a dataset using the populated data classes.
+
+    Parameters
+    ----------
+    attr_mgr : ImapCdfAttributes
+        attribute manager used to get the data product field's attributes
+    logical_source: str
+        The logical source of the data product that's being created
+    data_fields: list[dataclasses.Field]
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all data product fields in xr.DataArray
+    """
+    # TODO: Once L1B DE processing is implemented using the spin packet
+    #  and relative L1A DE time to calculate the absolute DE time,
+    #  this epoch conversion will go away and the time in the DE dataclass
+    #  can be used direction
+    epoch_converted_time = [calc_start_time(1)]
+
+    # Create a data array for the epoch time
+    # TODO: might need to update the attrs to use new YAML file
+    epoch_time = xr.DataArray(
+        data=epoch_converted_time,
+        name="epoch",
+        dims=["epoch"],
+        attrs=attr_mgr.get_variable_attributes("epoch"),
+    )
+
+    if logical_source == "imap_lo_l1c_pset":
+        esa_step = xr.DataArray(
+            data=[1, 2, 3, 4, 5, 6, 7],
+            name="esa_step",
+            dims=["esa_step"],
+            attrs=attr_mgr.get_variable_attributes("esa_step"),
+        )
+        pointing_bins = xr.DataArray(
+            data=np.arange(3600),
+            name="pointing_bins",
+            dims=["pointing_bins"],
+            attrs=attr_mgr.get_variable_attributes("pointing_bins"),
+        )
+
+        esa_step_label = xr.DataArray(
+            esa_step.values.astype(str),
+            name="esa_step_label",
+            dims=["esa_step_label"],
+            attrs=attr_mgr.get_variable_attributes("esa_step_label"),
+        )
+        pointing_bins_label = xr.DataArray(
+            pointing_bins.values.astype(str),
+            name="pointing_bins_label",
+            dims=["pointing_bins_label"],
+            attrs=attr_mgr.get_variable_attributes("pointing_bins_label"),
+        )
+        dataset = xr.Dataset(
+            coords={
+                "epoch": epoch_time,
+                "pointing_bins": pointing_bins,
+                "pointing_bins_label": pointing_bins_label,
+                "esa_step": esa_step,
+                "esa_step_label": esa_step_label,
+            },
+            attrs=attr_mgr.get_global_attributes(logical_source),
+        )
+
+    # Loop through the data fields that were pulled from the
+    # data class. These should match the field names given
+    # to each field in the YAML attribute file
+    for data_field in data_fields:
+        field = data_field.name.lower()
+        # Create a list of all the dimensions using the DEPEND_I keys in the
+        # YAML attributes
+        dims = [
```

Comment:
Nice. I like that you are extracting the dims form the attributes. I need to implement this in my pointing set code.

Reply:
I agree this is nice. I wonder if we should add this as a general helper to the AttributeManager class (not in this PR). `get_variable_dims()` (naming TBD) that would encapsulate this logic for everyone.